hardware/filters/filters.py

```diff
@@ -1,53 +1,171 @@
+        # attributes
+        self.attributes_table.add('Label', self.hardware.label)
+        self.attributes_table.add('Factor', self.hardware.factor)
+        self.scroll_layout.addWidget(self.attributes_table)
+        # mm input table
+        input_table = pw.InputTable()
+        input_table.add('Motor Position', None)
+        input_table.add('Current', self.hardware.motor_position)
+        self.motor_destination = self.hardware.motor_position.associate(display=False)
+        input_table.add('Destination', self.motor_destination)
+        self.scroll_layout.addWidget(input_table)
+        # set mm button
+        self.set_motor_button = pw.SetButton('SET POSITION')
+        self.scroll_layout.addWidget(self.set_motor_button)
+        self.set_motor_button.clicked.connect(self.on_set_motor)
+        g.queue_control.disable_when_true(self.set_motor_button)
+        # zero input table
+        input_table = pw.InputTable()
+        input_table.add('Zero Position', None)
+        input_table.add('Current', self.hardware.zero_position)
+        self.zero_destination = self.hardware.zero_position.associate(display=False)
+        input_table.add('Destination', self.zero_destination)
+        self.scroll_layout.addWidget(input_table)
+        # set zero button
+        self.set_zero_button = pw.SetButton('SET ZERO')
+        self.scroll_layout.addWidget(self.set_zero_button)
+        self.set_zero_button.clicked.connect(self.on_set_zero)
+        g.queue_control.disable_when_true(self.set_zero_button)
+        # horizontal line
+        self.scroll_layout.addWidget(pw.line('H'))
+        # home button
+        input_table = pw.InputTable()
+        self.home_button = pw.SetButton('HOME', 'advanced')
+        self.scroll_layout.addWidget(self.home_button)
+        self.home_button.clicked.connect(self.on_home)
+        g.queue_control.disable_when_true(self.home_button)
+        # finish
+        self.scroll_layout.addStretch(1)
+        self.layout.addStretch(1)
+        self.hardware.update_ui.connect(self.update)
+        
+    def on_home(self):
+        self.driver.hardware.q.push('home')
+        
+    def on_set_motor(self):
+        new_mm = self.motor_destination.read('deg')
+        self.hardware.set_motor_position(new_mm, units='deg')
+
+    def on_set_zero(self):
+        new_zero = self.zero_destination.read('deg')
+        self.driver.set_zero(new_zero)
+        self.driver.offset.write(0)
+        name = self.hardware.name
+        g.coset_control.read().zero(name)
+        self.driver.get_position()
+
+    def update(self):
+        pass
+
+
+# --- hardware ------------------------------------------------------------------------------------
 
 
 class Hardware(hw.Hardware):
     
     def __init__(self, *arks, **kwargs):
-        self.kind = 'filter'
+        self.kind = 'filter'        
+        self.factor = pc.Number(1, decimals=0)
+        self.motor_limits = pc.NumberLimits(min_value=-360., max_value=360., units='mm')
```

Comment:
Is there a reason that units are in `mm`?

Reply:
Because it never got changed to `deg` (after being blatantly stolen from delays)